### PR TITLE
Fix for napalm-automation/napalm#1145 = get_vlans() broken on NX-OS where no interfaces in VLAN

### PR DESCRIPTION
**Pull request by ggiesen**
Fix for napalm-automation/napalm#1145 = get_vlans() broken on NX-OS where no interfaces in VLAN

### CODE REVIEW

File: test/nxos/mocked_data/test_get_vlans/6000/expected_result.json [hunk 1]
```diff
@@ -1,44 +1,7 @@
 {
   "230": {
     "name": "database",
-    "interfaces": [
```
Comment (mirceaulinic) [inline test/nxos/mocked_data/test_get_vlans/6000/expected_result.json]: Could you please clarify why you're removing these?

I'm not familiar with this `vlanshowplist-ifidx`, from your description from #1145 I understand that `Using get_vlans for nxos_ssh fails when there are VLANs defined with no interfaces assigned.`, however in the mock data associated for this test case I understand that there are: https://github.com/napalm-automation/napalm/blob/8f986aaf4b2c1f7b8cbc5b27ac77ed78d7fdccbd/test/nxos/mocked_data/test_get_vlans/6000/show_vlan_brief.json#L17-L19... what's the secret key here? :smile:

Comment (ggiesen) [reply test/nxos/mocked_data/test_get_vlans/6000/expected_result.json]: vlanshowplist-ifidx is the list of Interfaces associated to the VLAN. If there are no interfaces associated to the VLAN, vlanshowplist-ifidx doesn't exist (which creates the bug). 

In order to create a test case for this, I just modified one VLAN from the existing test data to remove vlanshowplist-ifidx. You're right ithat I could have added a VLAN instead, let me know if that is your preference.

Comment (mirceaulinic) [reply test/nxos/mocked_data/test_get_vlans/6000/expected_result.json]: I see, yes, please add a separate test case, without adjusting the existing ones.

**Comment by mirceaulinic** [comment]
Bump @ggiesen

**Comment by ggiesen** [comment]
Added new test cases, however it looks like pylama is failing on code unrelated to my changes

File: test/nxos/mocked_data/test_get_vlans/interface_without_vlans/show_vlan_brief.json [hunk 2]
```diff
@@ -0,0 +1,261 @@
+{
```
Comment (mirceaulinic) [inline test/nxos/mocked_data/test_get_vlans/interface_without_vlans/show_vlan_brief.json:1]: It looks like every VLAN in this output has the `vlanshowplist-ifidx`?

File: test/nxos/mocked_data/test_get_vlans/interface_without_vlans/show_vlan_brief.json [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        "vlanshowbr-shutstate": "noshutdown",
+        "vlanshowplist-ifidx": "port-channel12,port-channel100,Ethernet1/21,Ethernet1/22,Ethernet1/23,Ethernet1/24,Ethernet1/47,Ethernet1/48"
+      },
+      {
+        "vlanshowbr-vlanid": "1115",
+        "vlanshowbr-vlanid-utf": "1115",
+        "vlanshowbr-vlanname": "Vlan1115",
+        "vlanshowbr-vlanstate": "active",
+        "vlanshowbr-shutstate": "noshutdown"
+      },
```
Comment (ggiesen) [inline test/nxos/mocked_data/test_get_vlans/interface_without_vlans/show_vlan_brief.json:226]: VLAN 1115 does not have vlanshowplist-ifidx. I can trim out some of the extraneous VLANs/add a few more VLANs without interfaces to make the test case more apparent if that would be better

Comment (mirceaulinic) [reply test/nxos/mocked_data/test_get_vlans/interface_without_vlans/show_vlan_brief.json:226]: Na, that's fine, I missed it. Thanks for clarifying!

**Pull request merged**